wa/workloads/gfxbench/__init__.py

```diff
@@ -0,0 +1,77 @@
+#    Copyright 2014-2016 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import re
+
+from wa import ApkUiautoWorkload, WorkloadError, Parameter
+
+
+class Gfxbench(ApkUiautoWorkload):
+
+    name = 'gfxbench-corporate'
+    package_names = ['net.kishonti.gfxbench.gl.v50000.corporate']
+    regex_matches = [re.compile(r'Car Chase score (.+)'),
+                     re.compile(r'Car Chase Offscreen score (.+)'),
+                     re.compile(r'Manhattan 3.1 score (.+)'),
+                     re.compile(r'1080p Manhattan 3.1 Offscreen score (.+)'),
+                     re.compile(r'1440p Manhattan 3.1 Offscreen score (.+)'),
+                     re.compile(r'Tessellation score (.+)'),
+                     re.compile(r'Tessellation Offscreen score (.+)')]
+    description = '''
+    Execute a subset of graphical performance benchmarks
+
+    Test description:
+    1. Open the gfxbench application 
+    2. Execute Car Chase, Manhattan and Tessellation benchmarks
+
+    '''
+
+    parameters = [
+        Parameter('subtest_x', kind=int, default=840,
+                  description="""
+                  The x co-ordinate of the test selection menu required
+                  """),
+        Parameter('subtest_y', kind=int, default=1200,
+                  description="""
+                  The y co-ordinate of the test selection menu required
+                  """)
```

Comment:
Just noticed these. Why are you doing it this way, rather than specifying the names of the sub-tests to run? The coordinates are going to  vary all the time, and it's not something that the user of the test should need to worry about.

Reply:
There are no element names given for this menu. This is the only way to do it as far as I can see.

Reply:
Presumably, the entries in the menu are still distinct elements? So you can still select them based on their relative position to each other within the menu? You'd have to maintain a name --> position index mapping your code, in that case, but that would still be much preferable to expecting the user to calculate the coordinates each time.

Reply:
They are not. 

![image](https://user-images.githubusercontent.com/25177567/39997774-f4aff7f6-577b-11e8-96aa-2ffbdbcc595e.png)

Reply:
OK what about reading the bounds of the object  with id `main_circleControl` and then calculating the x/y coordinates as offsets from its center? You'd need to check how this elem scales in hoziontal mode, but if its in anyway sensible, calculating the offset from the center of control as a proportion of its total size should work.